test/fast/Unit tests/nvm_die_on_prefix

```diff
@@ -75,7 +75,7 @@ OUTPUT="$(nvm_die_on_prefix 0 foo "$(nvm_version_dir new)" 2>&1)"
 EXPECTED_OUTPUT="nvm is not compatible with the npm config \"prefix\" option: currently set to \"./bad prefix\"
 Run \`npm config delete prefix\` or \`foo\` to unset it."
 EXIT_CODE="$(nvm_die_on_prefix 0 foo "$(nvm_version_dir new)" >/dev/null 2>&1; echo $?)"
-[ "_$OUTPUT" = "_$EXPECTED_OUTPUT" ] || die "'nvm_die_on_prefix 0 foo' did not error with '$EXPECTED_OUTPUT' with bad prefix set; got '$OUTPUT'"
+[[ "_$OUTPUT" == _$EXPECTED_OUTPUT* ]] || die "'nvm_die_on_prefix 0 foo' did not error with '$EXPECTED_OUTPUT' with bad prefix set; got '$OUTPUT'"
```

Comment:
this is bash-only syntax, and tests have to run everywhere nvm does, which includes non-bash POSIX shells.

Reply:
I reverted this fix as it was out of scope of the issues I was fixing for this story. The ticket about the 6 failing macOS tests would be a better place for it